src/rebar_utils.erl

```diff
@@ -414,8 +414,14 @@ user_agent() ->
     ?FMT("Rebar/~s (OTP/~s)", [Vsn, otp_release()]).
 
 reread_config(ConfigList) ->
+    SetEnv = case version_tuple(?MODULE:otp_release()) of
+        {X, _, _} when X =< 17 ->
+            fun application:set_env/3;
+        _ ->
+            fun (App, Key, Val) -> application:set_env(App, Key, Val, [{persistent, true}]) end
+    end,
```

Comment:
See the comments above @talentdeficit , there's really no other way. Maybe try/catch, but that's even worse.